### PR TITLE
Dev security levels

### PR DESCRIPTION
**Pull request by BytesWelder**
Related to #75 

**Description of changes**
Added mixin for CBV sec settings and
a decorator for function based restrictions

**Notes**
Before merge we should remove the todo in IssuingCaTableView.
I left in there for demo purpose if other may want to see in action before merge.

**Legal** 
- [x] I certify that I have all necessary rights to publish this contribution under the MIT license. I agree to the Trustpoint CLA and have added my name to the AUTHORS.md file.

### CODE REVIEW

File: trustpoint/pki/views/issuing_cas.py [hunk 1]
```diff
@@ -25,13 +29,18 @@ class IssuingCaContextMixin(TpLoginRequiredMixin, ContextDataMixin):
...
-class IssuingCaTableView(IssuingCaContextMixin, TpLoginRequiredMixin, SingleTableView):
+# @TODO: Remove Security Level Restriction from IssuingCaTableView. Only for demo purpose
+class IssuingCaTableView(SecurityLevelMixin, IssuingCaContextMixin, TpLoginRequiredMixin, SingleTableView):
     """Issuing CA Table View."""
 
     model = IssuingCaModel
     table_class = IssuingCaTable
     template_name = 'pki/issuing_cas/issuing_cas.html'
+    security_feature = SecurityFeatures.ISSUING_CA_TABLE_VIEW
+    no_permisson_url = '/home/dashboard/'
```
Comment (Aircoookie) [inline trustpoint/pki/views/issuing_cas.py:40]: Would you agree in renaming this to `disabled_by_security_level_url`? `no_permission_url` implies user-level permission for me personally. Unless we want to implement user-level access permissions at a later time and use the same URL for both security level and user role based redirects.

File: trustpoint/sysconf/views.py [hunk 2]
```diff
@@ -16,6 +21,79 @@
...
+        --------
+        str
+            The security mode of the current security level instance.
+        """
+        return cls._get_security_level_instance().security_mode
+
+    @classmethod
+    def refresh_security_level_instance(cls):
+        """Refreshes the cached security level instance by reloading it from the database."""
+        cls._thread_locals.security_level_instance = SecurityConfig.objects.first()
```
Comment (Aircoookie) [inline trustpoint/sysconf/views.py:73]: Doing this using threading locals seems like a good way to implement a singleton, as long as we can guarantee that all views are called on the same thread. An alternative might be https://python-patterns.guide/gang-of-four/singleton/ , however that seems to not work well with `__init__()`, which may or may not be required for Mixins.

File: trustpoint/pki/views/issuing_cas.py (hunk 1 above)
Comment (BytesWelder) [reply trustpoint/pki/views/issuing_cas.py]: Fully.
As you said:
There are only two hard things in Computer Science: cache invalidation and naming things.

Done

**Comment by BytesWelder** [comment]
@Aircoookie @AlexHx8472 
I moved the get_security_level from the mixin to to security_manager.
Should be cleaner this way but please review it.

If approved, ready to merge.

**Review by Aircoookie — APPROVED** [review]

**Pull request merged**